src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4452,15 +4452,8 @@ protected virtual void Dispose(bool disposing)
                 catch (Exception exception) when (!ExceptionCheck.IsFatal(exception)) { }
             }
 
-            // Make sure we're the first call to Dispose and no SetAsyncEventSelect is in progress.
-            int last;
-            SpinWait sw = new SpinWait();
-            while ((last = Interlocked.CompareExchange(ref _intCleanedUp, 1, 0)) == 2)
-            {
-                sw.SpinOnce();
-            }
-
-            if (last == 1)
```

Comment:
I agree based on the current code this change is goodness.  But do you know what change caused that?  Was it never needed in the .NET Core codebase, or was there a PR that removed other settings of _intCleanedUp?

Reply:
Thanks.  Makes me wonder what race condition it was trying to prevent and whether there's an issue in core as a result.

Reply:
I'm looking a bit further at the code. I guess it deals with WSAEventSelect failing when an async connect/accept is beginning while the Socket is disposed. Then the Dispose method will complete those operations. I'm only guessing though... The WSAEventSelect is not used in Windows .NET Core Sockets, so this is implemented differently.

Reply:
Thanks for looking.